### DISCUSSION TITLE
Q2S2 (Mid-May to June) Roadmap Planning

### DISCUSSION THREAD

**Discussion by BenJam** [opencollective/opencollective#4273] [Ideas]
It's the start of a new sprint! I felt like we did pretty well last sprint with a discussion, and with last week's async team retreat I thoguht we should continue to be inclusive and respectful of the distributed nature of the team. 

## What is this discussion for?
Prioriting [projects](https://github.com/opencollective/opencollective/issues?q=is%3Aopen+is%3Aissue+label%3Aproject) that Open Collective will dedicate our time to weeks commencing Mon 17th May to 25th June.

## How do I contribute?

**Pitching a new project**
Create a new project-level issue for it (which should help organise thoughts) and briefly introduce it in a top-level comment. Make sure you include your case as to _why_ we should commit ourselves to the idea, what benefit you expect to see and how we could measure our success.

**Pitching a project that has already been scoped**
Create a top-level comment introduce your case for a project. Link to the project issue(s) that are appropriate. Consider `@mention`ing the people who created the original issues to get their opinion. Look to the roadmap (see below) for ideas that were pitched previously. 

**Pitching a project that has already begun**
In some cases (ledger refactor) we are already part-way through a project. In which case I would encourage you to consider what can be achieved in the next six weeks, what we can celebrate at the end of the sprint and whether we can release anything that will help us check an assumption or mitigate a risk.  

**Pitching other investments that are not a 'project'**
Is completely legitimate too, just create a top-level comment for it. Try to ask yourself the same _why_questions so that others can compare it with time we might dedicate to projects: 'why do this', 'why now?', 'what do we expect to see a result of this work?' and 'how will we gauge our success?' are all valid questions to consider. 

**Discussing others' pitches**
Feel free to lend thoughts, opionions, data and ask questions in threads from each of the top-level comments (which should contain pitch). 

**Voting**
Hit the upvote on ideas that resonate with you, try to limit yourself to three to five votes, we can only do so much each sprint. 

## Our Priorities
It's obvious but worth repeating that we should prioritise projects that align with our mission _and_ our priorities at the time we commit to a project. At the Open Collective team 'retreat' we reaffirmed our mission and identified our priorities for Open Collective in 2021:

> Our goal is to catalyze impact by enabling communities to have economic power, providing them with tools to be financially independent and unlock access to funding.

Our priorities for 2021 are about deepening our core strengths: 

* Building up our own non-profit Fiscal Hosts and scaling up adoption by third party Fiscal Hosts
* Facilitating more contributions to Collectives
* Facilitating disbursement of funds.

## [Our Roadmap](https://github.com/opencollective/opencollective/projects/54)
When we ran this process six weeks ago we created a project board (link in header) that contained every idea that was presented. Some of those ideas were acted on immediately, evreything else was punted down the road.

I'd like us to consider the priority of everything in the roadmap relative to the priorities that we discuss here. So feel free to create a project issue for anything in the roadmap, and [re-present anything here that you proposed last time](https://github.com/opencollective/opencollective/discussions/4161) with a project issue. 

## Next steps
**Contributing on loom**
In order to create space for those not centred around CET timezone I thought we might encourage others to pre-record their pitches, inclusive of any feedback and/or discussions that you'd had prior to the synchronous discussion.

**Synchronous discussion**
We'll arrange a synchronous discussion for everyone Monday 17th to talk through the top pitches for the quarter. From there we'll scope a sprint's worth of work.

**Kickoffs and further discussion**
The remainder of the week will be dedicated to scoping a six-week effort from a project. I would encourage everyone to contribute to as many discussions as possible.  The goal is that, at the end of the week, we understand 

- what the goal for the sprint is, 
- who will need to contribute, when and how long we expect that to take
- (broadly) the solution and/or ux that we're going to build to deliver it
- the boundaries for the scope

The key here is going to be time management. You know how much time you have avaiaible, when you're likely to have the time to contirbute and how long you might need to contribute it. If there's an obvious contention (you're needed in two places at once) voice it and we'll work something out.

**Comment by BenJam** [comment]
## All your [meta]base: levelling up our data capabilities

I'm kicking off this discussion with an issue I have just created about our data in metabase https://github.com/opencollective/opencollective/issues/4282

Understanding how the work we do impats the metrics that matter to us is key to creating a better product and a more focussed team. I would _love_ us to incorporate a data-centred retrospective into the project process in which we ask ourseles whether we achived what we expected to achieve, whether we measured the right thing and whether we should do more of the same. 

I am _certian_ that many of the questions linked in the issue above are already answered in Metabase, but I've been unable to readily find them. I'd love for _everyone_ in the team to level up on metabase, so that they can formulate their own plans around project impact and success metrics, but I am more than happy to run point on this immediately and offer some support to others looking to scope projects and determine how to measure the impact. 

If I could spend some time next week scoping with someone who had a deep level of understanding of our data and how we'e configured Metabase to reflect it I think I might be able to scope this down to a very small (<1 week) project.

**Reply by piamancini** [reply]
+1 to this

**Reply by patcon** [reply]
+1 to metabase! Really rad BI tool, and very great community. Embeddable widgets are stellar. ~~Would need to get graphql connector working (metabase doesn't support graphql yet! https://github.com/metabase/metabase/issues/6297), or convert to another connector type~~ Errrr postgres works 🤦

**Reply by timfong888** [reply]
How open is access to the queries on meta base?  Are they limited in access and scope?

**Reply by patcon** [reply]
> How open is access to the queries on meta base? Are they limited in access and scope?

Metabase is known for their expressive visual query builder, but they also allow using raw SQL and dropdowns for some token replacement, and "SQL snippets" for reuse of SQL across different "questions": https://www.metabase.com/docs/latest/users-guide/writing-sql.html

EDIT: These queries are against metabase's internal representation fwiw

**Comment by alanna** [comment]
# Improve Events #4280

**Reply by patcon** [reply]
Gather.Town all the things! Happy to share and help on this :)

Demo: https://link.civictech.ca/gathertown

**Comment by alanna** [comment]
# Privacy/Transparency #4214

**Reply by piamancini** [reply]
I think this is important and i'd like to make progress. It's unclear still what exactly this would entail. 
What are the top 3 issues here to solve? What features need scoping?

**Reply by Memo-Es** [reply]
I wanted to propose to implement https://github.com/opencollective/opencollective/issues/3998 for this sprint, but I saw it under this project and it made sense to tackle this as part of this big initiative.

It seems that if we want to tackle this for the next sprint we should aim for do a proper discussion and scoping and maybe 2 or 3 main goals to start moving forward.

**Comment by alanna** [comment]
# Crypto #4200

**Reply by piamancini** [reply]
This popped up again today, we should nail down a solution and just do it.

**Comment by Betree** [comment]
# Ledger refactoring https://github.com/opencollective/opencollective/issues/4124

This project is ongoing. We can already celebrate #4077 and the recording of `kind` for transactions, but a lot of the work is still to do. I however think that, given the right focus, we can close it within the next sprint. Things that are left to do:
- Record debt transactions for platform tips (work in progress)
- Adapt refunds for this new system (work in progress)
- Adapt the host settlement invoices to use this settlement system

**Reply by znarf** [reply]
I would add:
- Record Host Fee and Host Fee Share (+ debt) using the new system

**Reply by znarf** [reply]
Some proposed milestones:
- Be ready for Platform Tip invoicing on June 1st
- Be ready for Host Fee Share invoicing on July 1st

**Comment by BenJam** [comment]
## Forefronting Hosts

Adding a coupole of proposals from @alanna that look like they're quite well scoped and line up with a strategy that supports and places Hosts in the forefront:

- Easy acess to host terms: https://github.com/opencollective/opencollective/issues/2757
- Unhosting a collective: https://github.com/opencollective/opencollective/issues/3330

**Reply by piamancini** [reply]
This seems well scoped and low hanging fruit improvements for hosts.

**Comment by BenJam** [comment]
## Onboarding Improvements
I'd like to review the onboarding process for collectives, individuals, organisations and hosts and propose some simplifications.

Related issues 

- https://github.com/opencollective/opencollective/issues/3758
- https://github.com/opencollective/opencollective/issues/3901
- https://github.com/opencollective/opencollective/issues/4256

**Reply by alanna** [reply]
Can #4256 be part of this, or is it another thing?

**Reply by Memo-Es** [reply]
Can https://github.com/opencollective/opencollective/issues/3758 be implemented in this project?

**Reply by BenJam** [reply]
I've added those issues thanks, I think, given the conversation that we had about selecting 'who holds funds' I think it's worth looking at separating out the onboarding flows for collectives and hosts yes.

**Reply by kaylarep** [reply]
+1 for this...would love an overhaul of onboarding with really clear distinctions and steps for each topic. Also demo videos especially for non-tech savvy donors... To me this seems like a long-term, ongoing project, also would love to engage some user feedback (fresh eyes) to make sure it makes sense to someone in square 1.

**Comment by kewitz** [comment]
### (Transfer)Wise Secure Customer Authentication: OTT Refactor

This still only lives in my nightmare, I'll be creating a proper issue as soon as I spec it properly but beforehand I can already tell that it will require:
- Bundling transactions;
- Authorize Bundle payments using OTT;

**Reply by piamancini** [reply]
+1

**Reply by piamancini** [reply]
PS: you got this @kewitz !

**Reply by kewitz** [reply]
Project Issue: https://github.com/opencollective/opencollective/issues/4291

**Comment by Memo-Es** [comment]
## Search Experience Redesign https://github.com/opencollective/opencollective/issues/4295

Follow up of the project to design a better experience to find and apply to hosts on OC #4240

TL;DR
* Deprecate discover page 💀
* New search experience
* Better search capabilities

**Reply by kaylarep** [reply]
+1

Our search capabilities could be so much better. 
https://github.com/opencollective/opencollective/issues/2232
https://github.com/opencollective/opencollective/issues/3892

**Comment by Memo-Es** [comment]
## Implement Accessibility Enhancements https://github.com/opencollective/opencollective/issues/4045

Push the improvements described in the issue above to production; it may require split into 6 separate issues and tackled with different contributors using the engineering bounty program.

**Reply by Betree** [reply]
I should have definitely spent some time on this one for the last sprint. After a deeper review, I really think this can be done quickly: most of the issue is about updating values in our design system. I even wonder whether bounties are interesting for this since 95% of the issue is just to look at the website to make sure the new values look good.

I'd be happy to iterate on this project.

**Reply by kaylarep** [reply]
I put this in slack, but putting it here too: https://accessibility.blog.gov.uk/2016/09/02/dos-and-donts-on-designing-for-accessibility/

**Comment by Memo-Es** [comment]
## Implement Admin Dashboard https://github.com/opencollective/opencollective/issues/3880

Implement the new admin dashboard for all profiles. This new dashboard will merge the management dashboard features (host dashboard for hosts) with the settings.

Note: After implementation, all issues related with #2896 should be closed and considered as complete

**Comment by Betree** [comment]
# Product Communication Improvements #4213

More especially, I propose to implement the first element (_1. Changelog MVP_) from https://github.com/opencollective/opencollective/issues/4213#issuecomment-843205109: `Ability to publish a changelog entry` + `Implement the "What's new" modal`.

While the rest of the design is still being iterated on, this first block would give us the ability to beta-test the feature, which will help to prioritize further issues for this project. Plus, it should be fairly easy to implement, my estimate is around one or two days of work.

Ping @aminakazi, @Memo-Es

**Comment by Memo-Es** [comment]
## Contribution Flow Update https://github.com/opencollective/opencollective/issues/4299

An update on the contribution flow after one year of testing it live.

TL;DR

* Improve platform tips
* Contribution form redesign
* Layout enhancements

**Comment by piamancini** [comment]
Add structured bank data for Custom Payment method. I can't find the issue :(
Right now unless the host has transferwise connected, the payment info is stored in a free field box. 
We should let expense submitters include their info in a complete data form depending on their currency, even if it's a manual payment.

**Reply by kewitz** [reply]
https://github.com/opencollective/opencollective/issues/3893

**Comment by Memo-Es** [comment]
## Admin Dashboard - Reports Section https://github.com/opencollective/opencollective/issues/4300

Design the reports section of the admin dashboard for hosts

**Comment by Memo-Es** [comment]
## Clarificate Projects - Tiers - Subcollectives - Events With a Clear Design Proposal

I think that is what @alanna  mentions in https://github.com/opencollective/opencollective/issues/4281. We also had a discussion recently when we talked about a unified way to manage all these different budget capabilities.

I think this project can focus on having a clear scope and a preliminary design proposal to discuss over.

**Reply by BenJam** [reply]
Prioritisation notes: goal to provide greater clarity on how to manage budgets. Next step, discussion on what our intention should be for this and ut forward a proposal for next sprint.

**Comment by znarf** [comment]
### Virtual Cards

Follow up?

- Better limits
- Better auto-pause
- Improve Upload Receipt flow and reminders

**Reply by BenJam** [reply]
prioritisation notes: will continue to push for adoption, need to deal with the pains for the moment of auto pause being a little OTT, limits may cause problems, not uploading reciepts etc. @BenJam to start logging feedback in a seperate issue.  Quick fix: billing address for virtual cards, name on card? Next steps: monitor.

**Reply by kaylarep** [reply]
need to implement a stop mechanism

**Comment by aminakazi** [comment]
### Clarify difference between Updates and conversations
One way to do it: They could lay under one umbrella of 'Connect'.
Partially merge them (by keeping their individual functionalities, but changing how user's access it) 

One minute loom to explain my thoughts better: https://www.loom.com/share/5cdd6313425148fdb45a99d66f83a419?sharedAppSource=team_library

Reason: Having updates and conversations as separate sections results in confusion about the purpose of each, especially with the new updates UI.

**Reply by Betree** [reply]
- This new page will group updates and conversations. Do we want to get rid of `/collective/conversations`?
- What does that mean for the collective page, do we want to merge the sections there as well and apply the same pattern of having a button to switch between updates and conversations?

I think the intent of this project, clarifying the difference between updates and conversations, is good. I wonder if merging is the best way to achieve that.

**Reply by aminakazi** [reply]
We can absolutely discuss it further if we pick it up this sprint. Also, I'll rename the project to 'Clarify difference between updates and conversations', its more straightforward. Thanks :)

**Reply by kaylarep** [reply]
Merging them makes sense to me. 

What about naming the umbrella "Contact" instead?  "Connect" is pretty, but a little vague

**Comment by kewitz** [comment]
## Donating with Digital Wallets

Last-minute suggestion to sync with our annual priorities.
It should be relatively simple to implement support for paying with Apple Pay, Ali Pay, Google Pay, WeChat Pay, and other digital wallets through Stripe.
The last time I specced this, we were just missing some webhook infrastructure and the payment business logic.

**Reply by BenJam** [reply]
would be good to prototype this sprint, and look to support this the following... goals would be to scope, explore implementation, de risk etc.

**Comment by timfong888** [comment]
Hi I meant will there be user access and what it the scope permitted by a
given user?  For example, can I query across all projects even if I am not
an admin?

On Wed, May 26, 2021 at 10:25 PM Patrick Connolly ***@***.***>
wrote:

> How open is access to the queries on meta base? Are they limited in access
> and scope?
>
> Metabase is known for their expressive visual query builder, but they also
> allow using raw SQL and dropdowns for some token replacement, and "SQL
> snippets" for reuse of SQL across different "questions":
> https://www.metabase.com/docs/latest/users-guide/writing-sql.html
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/opencollective/opencollective/discussions/4273#discussioncomment-790265>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAADNZ334WLWO5N5XCPLERDTPXJUHANCNFSM44VNMEUQ>
> .
>
-- 

Inline image

Tim Fong

p: 415-508-6888

e: ***@***.***
